compiler/optimizer/GlobalRegisterAllocator.cpp

```diff
@@ -800,29 +800,35 @@ TR_GlobalRegisterAllocator::restoreOriginalSymbol(TR::Node *node, vcount_t visit
             {
             TR_RegisterCandidate *origRc = _registerCandidates[origSymRef->getReferenceNumber()];
 
-            if (setValueModified)
-               _valueModifiedSymRefs->set(origRc->getSymbolReference()->getReferenceNumber());
+            if (origRc)
```

Comment:
I think I'd rather pull this out of the current pull request - it seems a lot more complicated and worthy of separate consideration.